### PR TITLE
THRIFT-5633: move xenial to old, deprecate xenial in travis build

### PR DESCRIPTION
**Pull request by Jimexist**
- move xenial to old
- deprecate xenial in travis build

- [x] Did you create an [Apache Jira](https://issues.apache.org/jira/projects/THRIFT/issues/) ticket?  (not required for trivial changes)
- [x] If a ticket exists: Does your pull request title follow the pattern "THRIFT-NNNN: describe my issue"?
- [x] Did you squash your changes to a single commit?  (not required, but preferred)
- [x] Did you do your best to avoid breaking changes?  If one was needed, did you label the Jira ticket with "Breaking-Change"?
- [x] If your change does not involve any code, include `[skip ci]` anywhere in the commit message to free up build resources.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -58,12 +58,6 @@ env:
 jobs:
   include:
     # ========================= stage: docker =========================
-    - stage: docker
```
[the remark this answers is not in the document]
Comment (Jimexist) [reply .travis.yml:61]: yes i now kept the stage line; the env and script were mismatched due to git diff.

**Comment by Jimexist** [comment]
to add jammy see #2684

**Comment by Jimexist** [comment]
> It doesn't seem the travis issues are caused by this change (but I don't remember seeing the c++ and python test failures before 🤔)
> 
> also a minor thing that can totally be done in another PR after this: currently since we support 2 go versions, we use xenial for the older version and bionic+focal for the newer version:
> 
> * https://github.com/apache/thrift/blob/4f72788418d8ee331d8841310a4066fd0ba2e510/build/docker/ubuntu-xenial/Dockerfile#L139
> * https://github.com/apache/thrift/blob/4f72788418d8ee331d8841310a4066fd0ba2e510/build/docker/ubuntu-bionic/Dockerfile#L147
> * https://github.com/apache/thrift/blob/4f72788418d8ee331d8841310a4066fd0ba2e510/build/docker/ubuntu-focal/Dockerfile#L148
> 
> now with xenial going away, we would need to shift bionic (which would be the new oldest version we run on travis) to the older go version in order to still test both.

hi @fishy how do you feel is that testing 1.18 is done in #2697 (as a follow up), then we don't need to update in here?

**Comment by fishy** [comment]
sure as I said that can totally be done in a follow up pr instead.

**Pull request merged**